src/sentry/static/sentry/app/views/organizationEventsV2/transactionView/minimap.tsx

```diff
@@ -0,0 +1,663 @@
+        spanID: spanChild.span_id,
+        lookup,
+        pickSpanBarColour,
+        generateBounds,
+        span: spanChild,
+      });
+
+      acc.push(<React.Fragment key={key}>{results}</React.Fragment>);
+
+      return acc;
+    }, []);
+
+    return (
+      <React.Fragment>
+        <MinimapSpanBar
+          style={{
+            backgroundColor: spanBarColour,
+            left: spanLeft,
+            width: spanWidth,
+          }}
+        />
+        {reduced}
+      </React.Fragment>
+    );
+  };
+
+  render() {
+    return (
+      <MinimapBackground>
+        <BackgroundSlider id="minimap-background-slider">
+          {this.drawMinimap()}
+        </BackgroundSlider>
+      </MinimapBackground>
+    );
+  }
+}
+
+const TimeAxis = styled('div')`
+  width: 100%;
+  position: absolute;
+  left: 0;
+  top: ${MINIMAP_HEIGHT}px;
+
+  border-top: 1px solid #d1cad8;
+
+  height: ${TIME_AXIS_HEIGHT}px;
+  background-color: #faf9fb;
+
+  color: #9585a3;
+  font-size: 10px;
+  font-weight: 500;
+`;
+
+const TickLabelContainer = styled('div')`
+  height: ${TIME_AXIS_HEIGHT}px;
+
+  position: absolute;
+  top: 0;
+
+  user-select: none;
+`;
+
+const TickText = styled('span')`
+  line-height: 1;
+
+  position: absolute;
+  bottom: 8px;
+  white-space: nowrap;
+
+  ${({align}: {align: TickAlignment}) => {
+    switch (align) {
+      case TickAlignment.Center: {
+        return 'transform: translateX(-50%)';
+      }
+      case TickAlignment.Left: {
+        return null;
+      }
+
+      case TickAlignment.Right: {
+        return 'transform: translateX(-100%)';
+      }
+
+      default: {
+        throw Error(`Invalid tick alignment: ${align}`);
+      }
+    }
+  }};
+`;
+
+const TickMarker = styled('div')`
+  width: 1px;
+  height: 5px;
+
+  background-color: #d1cad8;
+
+  position: absolute;
+  top: 0;
+  left: 0;
+
+  transform: translateX(-50%);
+`;
+
+const TickLabel = (props: {
+  style: React.CSSProperties;
+  hideTickMarker?: boolean;
+  align?: TickAlignment;
+  duration: number;
+}) => {
+  const {style, duration, hideTickMarker = false, align = TickAlignment.Center} = props;
+
+  return (
+    <TickLabelContainer style={style}>
+      {hideTickMarker ? null : <TickMarker />}
+      <TickText align={align}>{getHumanDuration(duration)}</TickText>
+    </TickLabelContainer>
+  );
+};
+
+const DurationGuideBox = styled('div')`
+  position: absolute;
+
+  background-color: rgba(255, 255, 255, 1);
+  padding: 4px;
+
+  border-radius: 3px;
+  border: 1px solid rgba(0, 0, 0, 0.1);
+
+  height: 16px;
+
+  line-height: 1;
+  vertical-align: middle;
+
+  transform: translateY(50%);
+
+  white-space: nowrap;
+
+  ${({alignLeft}: {alignLeft: boolean}) => {
+    if (!alignLeft) {
+      return null;
+    }
+
+    return 'transform: translateY(50%) translateX(-100%) translateX(-8px);';
+  }};
+`;
+
+const MinimapContainer = styled('div')`
+  width: 100%;
+  position: sticky;
+  left: 0;
+  top: 0;
+  z-index: 99999999999;
```

Comment:
Could we use `theme.zIndex` here?

Reply:
I've decided to collate the z-indices locally https://github.com/getsentry/sentry/pull/14225/commits/3fe6709d60e3d9c17981f6080bca08bcdf55671b